source/extensions/filters/network/postgres_proxy/postgres_decoder.cc

```diff
@@ -241,16 +258,26 @@ bool DecoderImpl::onData(Buffer::Instance& data, bool frontend) {
     }
   }
 
-  std::vector<MsgAction>& actions = std::get<1>(msg.get());
+  std::vector<MsgAction>& actions = std::get<2>(msg.get());
   for (const auto& action : actions) {
     action(this);
   }
 
   ENVOY_LOG(debug, "({}) command = {} ({})", msg_processor.direction_, command_,
             std::get<0>(msg.get()));
-  ENVOY_LOG(debug, "({}) length = {}", msg_processor.direction_, getMessageLength());
-  ENVOY_LOG(debug, "({}) message = {}", msg_processor.direction_, getMessage());
+  ENVOY_LOG(debug, "({}) length = {}", msg_processor.direction_, message_len_);
+  if (ENVOY_LOG_CHECK_LEVEL(debug)) {
+    auto f = std::get<1>(msg.get());
+    std::string message = "Unrecognized";
+    if (f != nullptr) {
+      auto msgParser = f();
+      msgParser->read(data, message_len_ - 4);
```

Comment:
Mind to put a comment on `- 4` here?

Reply:
Added the comment explaining why 4 bytes have to be subtracted here.